daemon/api_validate.go

```diff
@@ -316,6 +351,38 @@ func forgetValidationSet(st *state.State, accountID, name string, sequence int)
 	return SyncResponse(nil, nil)
 }
 
+func validationSetAssertFromDb(st *state.State, accountID, name string, sequence int) (*asserts.ValidationSet, error) {
+	headers := map[string]string{
+		"series":     release.Series,
+		"account-id": accountID,
+		"name":       name,
+		"sequence":   fmt.Sprintf("%d", sequence),
+	}
+	st.Lock()
+	db := assertstate.DB(st)
+	st.Unlock()
+	as, err := db.Find(asserts.ValidationSetType, headers)
+	if err != nil {
+		return nil, err
+	}
+	vset := as.(*asserts.ValidationSet)
+	return vset, nil
+}
+
+func validationSetForAssert(st *state.State, accountID, name string, sequence int) (*snapasserts.ValidationSets, error) {
+	st.Unlock()
+	as, err := validationSetAssertFromDb(st, accountID, name, sequence)
+	st.Lock()
+	if err != nil {
+		return nil, err
+	}
+	sets := snapasserts.NewValidationSets()
+	if err := sets.Add(as); err != nil {
+		return nil, err
+	}
+	return sets, nil
+}
+
 func validateAgainstStore(st *state.State, accountID, name string, sequence int, user *auth.UserState) Response {
```

Comment:
wondering now if this should for consistency look if the assertion was acked in the local db if is not found in the store?